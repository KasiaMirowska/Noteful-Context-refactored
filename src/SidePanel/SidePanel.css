.side-panel {
    flex: 1;
    /* border-right: 2px solid grey; */
    text-align: right;
  
    padding: 15px;
    margin: 10px
}

h2 {
    font-family: 'Barriecito', cursive;
    color: rgb(204, 80, 8);
    padding-left: 20px;
}
ul {
    list-style-type: none;
}
.side-button {
    width: 30%;
    font-family: 'Raleway', sans-serif;
    font-size: 15px;
    text-decoration: none;
    padding: 15px;
    margin: 10px;
    cursor: pointer;
  }