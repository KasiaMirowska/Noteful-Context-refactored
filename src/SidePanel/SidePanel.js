import React from 'react';
import NotefulContext from '../NotefulContext';
import './SidePanel.css';
import Folder from '../Folder/Folder';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

export default class SidePanel extends React.Component {
    static contextType = NotefulContext;

    render(){
        const folders = this.context.folders.map(folder => {
            return <Folder 
                    key={folder.id}
                    id={folder.id}
                    name={folder.name}
                    />
        })
        return (
            <div className='folder'>
                <h2>Note Folders</h2>
                <React.Fragment>
                {folders}
                </React.Fragment>
                <Link to='/newFolder'>
                    <button type='button' className='side-button'>
                        Add Folder
                    </button>
                </Link>
            </div>
        )
    };  
};

SidePanel.propTypes = {
    history: PropTypes.arrayOf(PropTypes.shape({
        length: PropTypes.number,
        action: PropTypes.string,
        location: PropTypes.object,
        createHref: PropTypes.func,
        push: PropTypes.func,
    })),
    location: PropTypes.arrayOf(PropTypes.shape({
        pathname: PropTypes.string,
        search: PropTypes.string,
        hash: PropTypes.string,
        key: PropTypes.string,
    })),
    match: PropTypes.arrayOf(PropTypes.shape({
        path: PropTypes.string,
        url: PropTypes.string,
        isExact: PropTypes.boolean,
        params : PropTypes.arrayOf(PropTypes.shape({
            path: PropTypes.string,
            url: PropTypes.string,
        }))
    })),
}

