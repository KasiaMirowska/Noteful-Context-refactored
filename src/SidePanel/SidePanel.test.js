import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import SidePanel from './SidePanel';
import { BrowserRouter as BR } from 'react-router-dom';



describe('SidePanel component', () => {
    
    it ('SidePanel renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<BR><SidePanel /></BR>, div)
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders the UI as expected', () => {
        const sidePanel = renderer.create(<BR><SidePanel /></BR>);

        expect(sidePanel.toJSON()).toMatchSnapshot();

    })

})