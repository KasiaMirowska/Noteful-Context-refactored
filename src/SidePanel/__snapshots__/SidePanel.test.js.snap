// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidePanel component renders the UI as expected 1`] = `
<div
  className="folder"
>
  <h2>
    Note Folders
  </h2>
  <a
    href="/newFolder"
    onClick={[Function]}
  >
    <button
      className="side-button"
      type="button"
    >
      Add Folder
    </button>
  </a>
</div>
`;
